frontend/src/concepts/pipelines/content/DeletePipelinesModal.tsx

```diff
@@ -0,0 +1,185 @@
+    deleteStatuses,
+    setDeleteStatuses,
+    onBeforeClose,
+    abortSignal,
+  } = useDeleteStatuses({
+    onClose,
+    type: 'pipeline',
+    toDeleteResources: [...toDeletePipelines, ...toDeleteVersions],
+  });
+  const resourceCount = toDeletePipelines.length + toDeletePipelineVersions.length;
+
+  if (resourceCount === 0) {
+    return null;
+  }
+
+  let deleteTitle;
+  let deleteName;
+  let deleteDescription = <>This action cannot be undone.</>;
+  if (resourceCount > 1) {
+    deleteTitle = 'Delete pipelines?';
+    if (toDeletePipelines.length === 0) {
+      deleteName = `Delete ${toDeletePipelineVersions.length} pipeline versions`;
+    } else if (toDeletePipelineVersions.length === 0) {
+      deleteName = `Delete ${toDeletePipelines.length} pipelines along with their versions`;
+      deleteDescription = (
+        <>All versions from {toDeletePipelines.length} pipelines will be deleted.</>
+      );
+    } else {
+      deleteName = `Delete ${getPipelineAndVersionDeleteString(
+        toDeletePipelines,
+        'pipeline',
+      )} along with ${
+        toDeletePipelines.length !== 1 ? 'their' : 'its'
+      } versions and ${getPipelineAndVersionDeleteString(toDeleteVersions, 'version')}`;
+      deleteDescription = (
+        <>
+          All versions from {getPipelineAndVersionDeleteString(toDeletePipelines, 'pipeline')} and{' '}
+          {getPipelineAndVersionDeleteString(toDeleteVersions, 'version')} from different pipelines
+          will be deleted.
+        </>
+      );
+    }
+  } else if (toDeletePipelineVersions.length === 1) {
+    deleteTitle = 'Delete pipeline version?';
+    deleteName = toDeletePipelineVersions[0].version.name;
+    deleteDescription = (
+      <>
+        <strong>{toDeletePipelineVersions[0].version.name}</strong>, a version of your{' '}
+        {`[${toDeletePipelineVersions[0].pipelineName}]`} pipeline, will be deleted.
+      </>
+    );
+  } else {
+    deleteTitle = 'Delete pipeline?';
+    deleteName = toDeletePipelines[0].name;
+  }
+
+  return (
+    <DeleteModal
+      title={deleteTitle}
+      isOpen={isOpen}
+      onClose={() => onBeforeClose(false)}
+      deleting={deleting}
+      error={error}
+      onDelete={() => {
+        if (resourceCount === 0) {
+          return;
+        }
+        setDeleting(true);
+        setError(undefined);
+
+        const allPromises = [
+          ...toDeletePipelines.map((resource) =>
+            api.deletePipeline({ signal: abortSignal }, resource.id),
+          ),
+          ...toDeletePipelineVersions.map((resource) =>
+            api.deletePipelineVersion({ signal: abortSignal }, resource.version.id),
+          ),
+        ];
+
+        if (allPromises.length === 1) {
+          allPromises[0]
+            .then(() => onBeforeClose(true))
+            .catch((e) => {
+              setError(e);
+              setDeleting(false);
+            });
+        } else {
+          // Promise.allSettled(allPromises)
```

Comment:
Yes I was testing this new implementation with the current implementation (Actually it's regarding to the deletion modal close issue you mentioned below), haven't found a solution yet. I think both implementations give the same result, I will choose to keep one of them (probably the `allSettled` one because it looks easier)